lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1683,7 +1683,7 @@ LLVMToSPIRVBase::transValueWithoutDecoration(Value *V, SPIRVBasicBlock *BB,
           }
         }
       }
-      BVarInit = transValue(Init, nullptr);
+      BVarInit = transValue(Init, nullptr, CreateForward, FuncTrans);
```

Comment:
I think that, rather than passing through the `CreateForward` and `FuncTrans` parameters, this should just always call it with `true` and `FuncTransMode::Pointer` here...